model/common/src/icon4py/model/common/orchestration/decorator.py

```diff
@@ -0,0 +1,673 @@
+            )(fuse_func)
+
+            # export DACE_ORCH_CACHE_FROM_DISK=(True or 1) if you want to activate it
+            cache_from_disk = get_env_bool("DACE_ORCH_CACHE_FROM_DISK", default=False)
+            dace_program_location = Path(
+                default_build_folder
+            ) / f"{fuse_func.__module__}.{fuse_func.__name__}".replace(".", "_")
+
+            if cache_from_disk and (Path(dace_program_location) / "program.sdfg").exists():
+                try:
+                    if self_name:
+                        self = compile_time_args_kwargs.pop(self_name)
+                        cache["sdfg"], _ = dace_program.load_sdfg(
+                            Path(dace_program_location) / "program.sdfg",
+                            self,
+                            **compile_time_args_kwargs,
+                        )
+                        (
+                            cache["compiled_sdfg"],
+                            _,
+                        ) = dace_program.load_precompiled_sdfg(
+                            dace_program_location, self, **compile_time_args_kwargs
+                        )
+                    else:
+                        cache["sdfg"], _ = dace_program.load_sdfg(
+                            Path(dace_program_location) / "program.sdfg",
+                            **compile_time_args_kwargs,
+                        )
+                        (
+                            cache["compiled_sdfg"],
+                            _,
+                        ) = dace_program.load_precompiled_sdfg(
+                            dace_program_location, **compile_time_args_kwargs
+                        )
+                except:  # noqa: E722
+                    raise ValueError(
+                        f"Corrupted cache. Remove `{default_build_folder}` folder and re-run the program."
+                    ) from None
+            else:
+                # If there is a cached folder delete it to avoid corrupted cache.
+                # At this point, we have already decided that we are not going to use it.
+                if dace_program_location.exists() and dace_program_location.is_dir():
+                    shutil.rmtree(dace_program_location)
+
+                if self_name:
+                    self = compile_time_args_kwargs.pop(self_name)
+                    cache["sdfg"] = dace_program.to_sdfg(
+                        self, **compile_time_args_kwargs, simplify=False, validate=True
+                    )
+                else:
+                    cache["sdfg"] = dace_program.to_sdfg(
+                        **compile_time_args_kwargs, simplify=False, validate=True
+                    )
+                sdfg = cache["sdfg"]
+
+                if sdfg.name != f"{fuse_func.__module__}.{fuse_func.__name__}".replace(".", "_"):
+                    raise ValueError(
+                        "fused_SDFG.name != {fuse_func.__module__}.{fuse_func.__name__}"
```

Comment:
I assume the intention was to use a format-string:
```suggestion
                        f"fused_SDFG.name != {fuse_func.__module__}.{fuse_func.__name__}"
```

Reply:
Good catch ;)